llama_index/vector_stores/typesense.py

```diff
@@ -0,0 +1,258 @@
+        client (Any): Typesense client
+        tokenizer (Optional[Callable[[str], List]]): tokenizer function.
+
+    """
+
+    stores_text: bool = True
+    is_embedding_query = False
+
+    def __init__(
+        self,
+        client: Any,
+        tokenizer: Optional[Callable[[str], List]] = None,
+        text_key: str = DEFAULT_TEXT_KEY,
+        collection_name: str = DEFAULT_COLLECTION_NAME,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+        metadata_key: str = DEFAULT_METADATA_KEY,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        import_err_msg = (
+            "`typesense` package not found, please run `pip install typesense`"
+        )
+        try:
+            import typesense  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        if client is not None:
+            if not isinstance(client, typesense.Client):
+                raise ValueError(
+                    f"client should be an instance of typesense.Client, "
+                    f"got {type(client)}"
+                )
+            self._client = cast(typesense.Client, client)
+        self._tokenizer = tokenizer or utils.globals_helper.tokenizer
+        self._text_key = text_key
+        self._collection_name = collection_name
+        self._collection = self._client.collections[self._collection_name]
+        self._batch_size = batch_size
+        self._metadata_key = metadata_key
+
+    @property
+    def client(self) -> Any:
+        """Return Typesense client."""
+        return self._client
+
+    @property
+    def collection(self) -> Any:
+        """Return Typesense collection."""
+        return self._collection
+
+    def _create_collection(self, num_dim: int) -> None:
+        fields = [
+            {"name": "vec", "type": "float[]", "num_dim": num_dim},
+            {"name": f"{self._text_key}", "type": "string"},
+            {"name": ".*", "type": "auto"},
+        ]
+        self._client.collections.create(
+            {"name": self._collection_name, "fields": fields}
+        )
+
+    def _create_upsert_docs(
+        self, embedding_results: List[NodeWithEmbedding]
+    ) -> List[dict]:
+        upsert_docs = []
+        for node in embedding_results:
+            doc = {
+                "id": node.id,
+                "vec": node.embedding,
+                f"{self._text_key}": node.node.text,
+                "ref_doc_id": node.ref_doc_id,
+                f"{self._metadata_key}": node_to_metadata_dict(node.node),
+            }
+            upsert_docs.append(doc)
+
+        return upsert_docs
+
+    @staticmethod
+    def _to_typesense_filter(standard_filters: MetadataFilters) -> str:
+        """Convert from standard dataclass to typesense filter dict."""
+        filters = {}
+        for filter in standard_filters.filters:
+            filters[filter.key] = filter.value
+
+        if "filter_by" in filters:
+            return str(filters["filter_by"])
```

Comment:
Will users be passing in a `filter_by` parameter directly in the [Typesense filter format](https://typesense.org/docs/0.24.1/api/search.html#filter-parameters)? 

Or if the intention is to convert from a different filter format that LlamaIndex uses to Typesense's filter format, you want to concatenate multiple filters using `&&` in Typesense.

Reply:
Ah sorry the intention is to have only one string the user can pass in under the "filter_by" which will be fed directly into  filter_by. Let me rewrite it to make things clearer.